### PR TITLE
fix(js): replace `deno_core` with `rquickjs`

### LINKED ISSUE #1690: feature: Use QuickJS instead of V8
Migrate from the current JS implementation to use QuickJS. 

**Technical Requirements**
- Use https://docs.rs/rquickjs/latest/rquickjs/
- Ensure all existing tests pass
- Perform a benchmark to compare differences in performance

**Additional Details**
- Here are the existing benchmarks https://bellard.org/quickjs/bench.html

### PR DESCRIPTION
**Pull request by mayant15**
**Summary:**
Replaces V8 with QuickJS as the JS runtime

**Issue Reference(s):** 
Fixes #1690 
/claim 1690

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [x] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Size Comparison

image

**After**
```
cargo bloat --crates --release -n 0

    Finished release [optimized] target(s) in 0.40s
    Analyzing target/release/tailcall

 File  .text     Size Crate
13.4%  21.0%   2.4MiB std
 6.6%  10.4%   1.2MiB tailcall
 3.4%   5.3% 612.6KiB [Unknown]
 2.7%   4.2% 482.8KiB hyper
 2.6%   4.1% 474.3KiB rustls
 2.4%   3.8% 442.2KiB h2
 2.4%   3.7% 428.0KiB async_graphql
 2.3%   3.6% 417.2KiB protobuf
 2.0%   3.1% 357.7KiB prost_reflect
 1.7%   2.6% 304.9KiB regex_automata
 1.6%   2.5% 285.0KiB tokio
 1.3%   2.1% 242.9KiB hashbrown
 1.3%   2.1% 240.1KiB regex_syntax
 1.2%   1.9% 216.1KiB serde_yaml
 1.2%   1.8% 209.7KiB opentelemetry_sdk
 1.0%   1.6% 183.9KiB serde_json
 1.0%   1.5% 176.4KiB clap_builder
 0.7%   1.1% 128.8KiB protox_parse
 0.7%   1.1% 124.2KiB http
 0.7%   1.1% 124.1KiB ring
 0.6%   1.0% 110.6KiB aho_corasick
 0.6%   0.9% 108.0KiB pest
 0.5%   0.9%  98.4KiB serde
 0.5%   0.7%  85.4KiB opentelemetry_stdout
 0.4%   0.7%  81.2KiB rquickjs_sys
 0.4%   0.7%  80.4KiB unsafe_libyaml
 0.4%   0.6%  74.5KiB moka
 0.4%   0.6%  73.2KiB reqwest
 0.4%   0.6%  72.9KiB tonic
 0.4%   0.6%  70.3KiB async_graphql_extension_apollo_tracing
 0.3%   0.5%  63.2KiB async_graphql_parser
 0.3%   0.5%  62.5KiB tracing_subscriber
 0.3%   0.5%  60.4KiB prost
 0.3%   0.5%  57.3KiB jsonwebtoken
 0.3%   0.5%  55.1KiB futures_util
 0.3%   0.5%  53.1KiB http_cache
 0.3%   0.4%  51.8KiB phonenumber
 0.3%   0.4%  50.5KiB opentelemetry
 0.3%   0.4%  50.5KiB serde_value
 0.3%   0.4%  49.6KiB nvml_wrapper_sys
 0.3%   0.4%  49.5KiB opentelemetry_proto
 0.3%   0.4%  48.5KiB url
 0.3%   0.4%  48.3KiB prometheus
 0.2%   0.4%  44.7KiB webpki
 0.2%   0.4%  40.8KiB async_graphql_value
 0.2%   0.3%  39.3KiB opentelemetry_otlp
 0.2%   0.3%  31.8KiB encoding_rs
 0.2%   0.3%  30.0KiB serde_path_to_error
 0.2%   0.2%  28.9KiB indexmap
 0.2%   0.2%  28.4KiB http_cache_semantics
 0.1%   0.2%  27.3KiB opentelemetry_prometheus
 0.1%   0.2%  26.5KiB inquire
 0.1%   0.2%  26.1KiB anyhow
 0.1%   0.2%  23.1KiB libmimalloc_sys
 0.1%   0.2%  22.4KiB sysinfo
 0.1%   0.2%  21.9KiB rquickjs_core
 0.1%   0.2%  21.5KiB prost_types
 0.1%   0.2%  20.9KiB chrono
 0.1%   0.2%  20.6KiB serde_qs
 0.1%   0.2%  17.5KiB crossterm
 0.1%   0.1%  16.1KiB libflate
 0.1%   0.1%  16.0KiB tower
 0.1%   0.1%  15.8KiB tokio_rustls
 0.1%   0.1%  15.2KiB idna
 0.1%   0.1%  14.9KiB rayon_core
 0.1%   0.1%  14.7KiB tracing_opentelemetry
 0.1%   0.1%  14.5KiB tracing_core
 0.1%   0.1%  13.8KiB http_cache_reqwest
 0.1%   0.1%  13.5KiB parking_lot
 0.1%   0.1%  12.6KiB base64
 0.1%   0.1%  11.4KiB bytes
 0.1%   0.1%  11.0KiB unicode_segmentation
 0.1%   0.1%  10.7KiB time
 0.1%   0.1%   9.4KiB dotenvy
 0.1%   0.1%   9.3KiB once_cell
 0.1%   0.1%   9.1KiB convert_case
 0.0%   0.1%   8.8KiB futures_channel
 0.0%   0.1%   8.7KiB schemars
 0.0%   0.1%   8.6KiB memchr
 0.0%   0.1%   8.6KiB futures_task
 0.0%   0.1%   8.4KiB rustls_pemfile
 0.0%   0.1%   8.3KiB crossbeam_channel
 0.0%   0.1%   8.0KiB sharded_slab
 0.0%   0.1%   7.3KiB rustls_pki_types
 0.0%   0.1%   7.1KiB futures_timer
 0.0%   0.1%   6.8KiB hyper_rustls
 0.0%   0.1%   6.0KiB futures_core
 0.0%   0.1%   5.9KiB http_serde
 0.0%   0.0%   5.8KiB regex
 0.0%   0.0%   5.7KiB htpasswd_verify
 0.0%   0.0%   5.5KiB ryu
 0.0%   0.0%   5.0KiB crossbeam_epoch
 0.0%   0.0%   4.9KiB ipnet
 0.0%   0.0%   4.9KiB opentelemetry_system_metrics
 0.0%   0.0%   4.8KiB opentelemetry_http
 0.0%   0.0%   4.7KiB httparse
 0.0%   0.0%   4.7KiB colored
 0.0%   0.0%   4.6KiB signal_hook
 0.0%   0.0%   4.4KiB unicode_normalization
 0.0%   0.0%   4.3KiB nom
 0.0%   0.0%   4.2KiB mime
 0.0%   0.0%   4.1KiB event_listener
 0.0%   0.0%   4.0KiB sha1
 0.0%   0.0%   3.9KiB regex_cache
 0.0%   0.0%   3.6KiB signal_hook_registry
 0.0%   0.0%   3.6KiB pwhash
 0.0%   0.0%   3.2KiB tracing
 0.0%   0.0%   3.1KiB nu_ansi_term
 0.0%   0.0%   3.1KiB bincode
 0.0%   0.0%   3.1KiB bcrypt
 0.0%   0.0%   3.0KiB webbrowser
 0.0%   0.0%   2.9KiB eyre
 0.0%   0.0%   2.7KiB libflate_lz77
 0.0%   0.0%   2.7KiB parking_lot_core
 0.0%   0.0%   2.5KiB md5
 0.0%   0.0%   2.5KiB mio
 0.0%   0.0%   2.4KiB anstream
 0.0%   0.0%   2.4KiB rustls_native_certs
 0.0%   0.0%   2.3KiB form_urlencoded
 0.0%   0.0%   2.3KiB blowfish
 0.0%   0.0%   2.1KiB anstyle
 0.0%   0.0%   1.9KiB reqwest_middleware
 0.0%   0.0%   1.9KiB protox
 0.0%   0.0%   1.9KiB hyper_timeout
 0.0%   0.0%   1.9KiB nvml_wrapper
 0.0%   0.0%   1.9KiB strsim
 0.0%   0.0%   1.9KiB crossbeam_deque
 0.0%   0.0%   1.8KiB thread_local
 0.0%   0.0%   1.8KiB cache_control
 0.0%   0.0%   1.8KiB percent_encoding
 0.0%   0.0%   1.8KiB sct
 0.0%   0.0%   1.7KiB security_framework
 0.0%   0.0%   1.7KiB tokio_util
 0.0%   0.0%   1.7KiB rayon
 0.0%   0.0%   1.6KiB smallvec
 0.0%   0.0%   1.6KiB tracing_log
 0.0%   0.0%   1.6KiB tokio_stream
 0.0%   0.0%   1.5KiB futures_executor
 0.0%   0.0%   1.5KiB httpdate
 0.0%   0.0%   1.5KiB opentelemetry_appender_tracing
 0.0%   0.0%   1.4KiB spin
 0.0%   0.0%   1.4KiB libloading
 0.0%   0.0%   1.4KiB triomphe
 0.0%   0.0%   1.3KiB async_lock
 0.0%   0.0%   1.2KiB event_listener_strategy
 0.0%   0.0%   1.2KiB crc32fast
 0.0%   0.0%   1.0KiB miette
 0.0%   0.0%   1.0KiB fast_chemail
 0.0%   0.0%     932B getrandom
 0.0%   0.0%     928B headers
 0.0%   0.0%     908B indenter
 0.0%   0.0%     856B core_foundation
 0.0%   0.0%     836B clap_lex
 0.0%   0.0%     788B uname
 0.0%   0.0%     652B tinyvec
 0.0%   0.0%     620B socket2
 0.0%   0.0%     620B serde_urlencoded
 0.0%   0.0%     600B linked_hash_map
 0.0%   0.0%     596B http_body
 0.0%   0.0%     548B quanta
 0.0%   0.0%     508B want
 0.0%   0.0%     408B rand
 0.0%   0.0%     396B parking
 0.0%   0.0%     332B slab
 0.0%   0.0%     324B ascii_utils
 0.0%   0.0%     268B headers_core
 0.0%   0.0%     260B zeroize
 0.0%   0.0%     248B tagptr
 0.0%   0.0%     184B log
 0.0%   0.0%     124B unicode_bidi
 0.0%   0.0%      96B untrusted
 0.0%   0.0%      20B subtle
 0.0%   0.0%       8B dyn_clone
63.7% 100.0%  11.3MiB .text section size, the file size is 17.8MiB

Note: numbers above are a result of guesswork. They are not 100% correct and never will be.
```

**Before**
```
cargo bloat --crates --release -n 0

    Finished release [optimized] target(s) in 0.24s
    Analyzing target/release/tailcall

 File  .text     Size Crate
27.6%  47.8%  14.4MiB v8
 8.5%  14.7%   4.4MiB [Unknown]
 4.8%   8.3%   2.5MiB std
 2.2%   3.9%   1.2MiB tailcall
 0.9%   1.6% 482.8KiB hyper
 0.9%   1.5% 474.3KiB rustls
 0.8%   1.5% 450.6KiB deno_core
 0.8%   1.4% 442.2KiB h2
 0.8%   1.4% 428.0KiB async_graphql
 0.8%   1.4% 417.2KiB protobuf
 0.7%   1.2% 357.3KiB prost_reflect
 0.6%   1.0% 304.9KiB regex_automata
 0.5%   0.9% 289.6KiB tokio
 0.5%   0.9% 275.3KiB hashbrown
 0.5%   0.8% 240.1KiB regex_syntax
 0.4%   0.7% 216.0KiB serde_yaml
 0.4%   0.7% 209.7KiB opentelemetry_sdk
 0.4%   0.6% 196.9KiB serde_json
 0.3%   0.6% 176.4KiB clap_builder
 0.2%   0.4% 128.8KiB protox_parse
 0.2%   0.4% 124.2KiB http
 0.2%   0.4% 124.1KiB ring
 0.2%   0.4% 116.8KiB serde
 0.2%   0.4% 110.6KiB aho_corasick
 0.2%   0.3% 106.1KiB pest
 0.2%   0.3%  85.4KiB opentelemetry_stdout
 0.2%   0.3%  80.4KiB unsafe_libyaml
 0.1%   0.2%  74.5KiB moka
 0.1%   0.2%  73.2KiB reqwest
 0.1%   0.2%  72.9KiB tonic
 0.1%   0.2%  70.3KiB async_graphql_extension_apollo_tracing
 0.1%   0.2%  64.9KiB async_graphql_parser
 0.1%   0.2%  62.4KiB tracing_subscriber
 0.1%   0.2%  60.4KiB prost
 0.1%   0.2%  57.3KiB jsonwebtoken
 0.1%   0.2%  55.3KiB futures_util
 0.1%   0.2%  53.1KiB http_cache
 0.1%   0.2%  51.8KiB phonenumber
 0.1%   0.2%  50.5KiB opentelemetry
 0.1%   0.2%  50.5KiB serde_value
 0.1%   0.2%  49.6KiB nvml_wrapper_sys
 0.1%   0.2%  49.5KiB opentelemetry_proto
 0.1%   0.2%  49.4KiB url
 0.1%   0.2%  48.3KiB prometheus
 0.1%   0.1%  44.7KiB webpki
 0.1%   0.1%  40.8KiB async_graphql_value
 0.1%   0.1%  39.3KiB opentelemetry_otlp
 0.1%   0.1%  34.6KiB anyhow
 0.1%   0.1%  31.8KiB encoding_rs
 0.1%   0.1%  30.0KiB serde_path_to_error
 0.1%   0.1%  28.4KiB http_cache_semantics
 0.1%   0.1%  27.3KiB opentelemetry_prometheus
 0.1%   0.1%  27.2KiB indexmap
 0.0%   0.1%  26.5KiB inquire
 0.0%   0.1%  23.1KiB sourcemap
 0.0%   0.1%  23.1KiB libmimalloc_sys
 0.0%   0.1%  22.4KiB sysinfo
 0.0%   0.1%  21.5KiB prost_types
 0.0%   0.1%  20.9KiB chrono
 0.0%   0.1%  20.6KiB serde_qs
 0.0%   0.1%  20.4KiB serde_v8
 0.0%   0.1%  17.5KiB crossterm
 0.0%   0.1%  16.1KiB libflate
 0.0%   0.1%  16.0KiB tower
 0.0%   0.1%  15.8KiB tokio_rustls
 0.0%   0.0%  15.2KiB idna
 0.0%   0.0%  14.9KiB rayon_core
 0.0%   0.0%  14.7KiB tracing_opentelemetry
 0.0%   0.0%  14.5KiB tracing_core
 0.0%   0.0%  13.8KiB http_cache_reqwest
 0.0%   0.0%  13.2KiB parking_lot
 0.0%   0.0%  12.6KiB base64
 0.0%   0.0%  11.4KiB bytes
 0.0%   0.0%  11.0KiB unicode_segmentation
 0.0%   0.0%  10.7KiB time
 0.0%   0.0%  10.7KiB futures_task
 0.0%   0.0%   9.6KiB once_cell
 0.0%   0.0%   9.5KiB futures_channel
 0.0%   0.0%   9.4KiB dotenvy
 0.0%   0.0%   9.1KiB convert_case
 0.0%   0.0%   8.7KiB schemars
 0.0%   0.0%   8.6KiB memchr
 0.0%   0.0%   8.4KiB rustls_pemfile
 0.0%   0.0%   8.3KiB crossbeam_channel
 0.0%   0.0%   8.0KiB sharded_slab
 0.0%   0.0%   7.3KiB rustls_pki_types
 0.0%   0.0%   7.1KiB futures_timer
 0.0%   0.0%   6.8KiB hyper_rustls
 0.0%   0.0%   6.0KiB futures_core
 0.0%   0.0%   5.9KiB http_serde
 0.0%   0.0%   5.8KiB regex
 0.0%   0.0%   5.7KiB htpasswd_verify
 0.0%   0.0%   5.5KiB ryu
 0.0%   0.0%   5.0KiB crossbeam_epoch
 0.0%   0.0%   4.9KiB ipnet
 0.0%   0.0%   4.9KiB opentelemetry_system_metrics
 0.0%   0.0%   4.8KiB opentelemetry_http
 0.0%   0.0%   4.7KiB httparse
 0.0%   0.0%   4.7KiB colored
 0.0%   0.0%   4.6KiB signal_hook
 0.0%   0.0%   4.4KiB unicode_normalization
 0.0%   0.0%   4.3KiB nom
 0.0%   0.0%   4.2KiB mime
 0.0%   0.0%   4.1KiB event_listener
 0.0%   0.0%   4.0KiB sha1
 0.0%   0.0%   3.9KiB regex_cache
 0.0%   0.0%   3.6KiB signal_hook_registry
 0.0%   0.0%   3.6KiB pwhash
 0.0%   0.0%   3.3KiB bincode
 0.0%   0.0%   3.2KiB tracing
 0.0%   0.0%   3.1KiB nu_ansi_term
 0.0%   0.0%   3.1KiB bcrypt
 0.0%   0.0%   3.0KiB webbrowser
 0.0%   0.0%   2.9KiB eyre
 0.0%   0.0%   2.7KiB libflate_lz77
 0.0%   0.0%   2.7KiB parking_lot_core
 0.0%   0.0%   2.6KiB uuid
 0.0%   0.0%   2.5KiB md5
 0.0%   0.0%   2.5KiB mio
 0.0%   0.0%   2.4KiB anstream
 0.0%   0.0%   2.4KiB rustls_native_certs
 0.0%   0.0%   2.3KiB form_urlencoded
 0.0%   0.0%   2.3KiB blowfish
 0.0%   0.0%   2.1KiB anstyle
 0.0%   0.0%   1.9KiB reqwest_middleware
 0.0%   0.0%   1.9KiB protox
 0.0%   0.0%   1.9KiB nvml_wrapper
 0.0%   0.0%   1.9KiB hyper_timeout
 0.0%   0.0%   1.9KiB strsim
 0.0%   0.0%   1.9KiB crossbeam_deque
 0.0%   0.0%   1.8KiB thread_local
 0.0%   0.0%   1.8KiB cache_control
 0.0%   0.0%   1.8KiB percent_encoding
 0.0%   0.0%   1.8KiB sct
 0.0%   0.0%   1.7KiB security_framework
 0.0%   0.0%   1.7KiB tokio_util
 0.0%   0.0%   1.7KiB rayon
 0.0%   0.0%   1.6KiB tracing_log
 0.0%   0.0%   1.6KiB smallvec
 0.0%   0.0%   1.6KiB tokio_stream
 0.0%   0.0%   1.5KiB futures_executor
 0.0%   0.0%   1.5KiB httpdate
 0.0%   0.0%   1.5KiB opentelemetry_appender_tracing
 0.0%   0.0%   1.4KiB spin
 0.0%   0.0%   1.4KiB libloading
 0.0%   0.0%   1.4KiB triomphe
 0.0%   0.0%   1.3KiB async_lock
 0.0%   0.0%   1.2KiB event_listener_strategy
 0.0%   0.0%   1.2KiB crc32fast
 0.0%   0.0%   1.0KiB miette
 0.0%   0.0%   1.0KiB fast_chemail
 0.0%   0.0%     932B getrandom
 0.0%   0.0%     928B headers
 0.0%   0.0%     908B indenter
 0.0%   0.0%     904B debugid
 0.0%   0.0%     856B core_foundation
 0.0%   0.0%     836B clap_lex
 0.0%   0.0%     788B uname
 0.0%   0.0%     652B tinyvec
 0.0%   0.0%     620B socket2
 0.0%   0.0%     620B serde_urlencoded
 0.0%   0.0%     600B linked_hash_map
 0.0%   0.0%     596B http_body
 0.0%   0.0%     548B quanta
 0.0%   0.0%     508B want
 0.0%   0.0%     408B rand
 0.0%   0.0%     396B parking
 0.0%   0.0%     376B bitvec
 0.0%   0.0%     332B slab
 0.0%   0.0%     324B ascii_utils
 0.0%   0.0%     268B headers_core
 0.0%   0.0%     260B zeroize
 0.0%   0.0%     248B tagptr
 0.0%   0.0%     184B log
 0.0%   0.0%     124B unicode_bidi
 0.0%   0.0%      96B untrusted
 0.0%   0.0%      20B subtle
 0.0%   0.0%       8B dyn_clone
57.8% 100.0%  30.0MiB .text section size, the file size is 51.9MiB

Note: numbers above are a result of guesswork. They are not 100% correct and never will be.
```

## Summary by CodeRabbit

- **New Features**
  - Integrated `rquickjs` for enhanced JavaScript runtime handling in Rust CLI applications, replacing `deno_core`.
  - Added conversion logic for `JsRequest`, `JsResponse`, and `Command` types to and from JavaScript values.
  - Modified the JavaScript `console` object to improve logging functionality.

- **Bug Fixes**
  - Updated comparison operations in `handle.rs` for more accurate file path handling.

- **Refactor**
  - Transitioned to `rquickjs` for JavaScript integration, affecting various components like requests, responses, and runtime management.

- **Tests**
  - Added comprehensive tests for new conversions between Rust types and JavaScript objects.

- **Documentation**
  - Enhanced inline comments and annotations for better clarity and maintenance of JavaScript integration code.

### CODE REVIEW

File: src/cli/javascript/request_filter.rs [hunk 1]
```diff
@@ -20,6 +21,34 @@ pub enum Command {
     Response(JsResponse),
 }
 
+impl<'js> FromJs<'js> for Command {
```
Comment (tusharmath) [inline src/cli/javascript/request_filter.rs:24]: Isn't there a macro to derive FromJS and ToJS?

File: src/cli/javascript/runtime.rs [hunk 2]
```diff
@@ -26,19 +30,34 @@ lazy_static::lazy_static! {
...
-        extension!(console, js = ["src/cli/javascript/shim/console.js",]);
-        let mut js_runtime = JsRuntime::new(RuntimeOptions {
-            extensions: vec![console::init_ops_and_esm()],
-            ..Default::default()
-        });
-        let global = js_runtime.execute_script("<anon>", FastString::from_static("globalThis"))?;
-        js_runtime.execute_script("<anon>", FastString::from(source))?;
-        tracing::debug!("JS Runtime created: {:?}", thread::current().name());
+        let js_runtime = rquickjs::Runtime::new().unwrap();
+        let context = Context::full(&js_runtime).unwrap();
```
Comment (tusharmath) [inline src/cli/javascript/runtime.rs:53]: drop the unwrap

File: src/cli/javascript/runtime.rs [hunk 3]
```diff
@@ -69,30 +88,40 @@ impl WorkerIO<Event, Command> for Runtime {
...
-            .map(|output| Ok(serde_v8::from_v8(scope, output)?))
-            .transpose()
+        runtime.context.with(|ctx| {
+            match event {
+                Event::Request(req) => {
+                    // NOTE: unwrap is safe here
+                    // We receive a `None` only if the name of the function is more than the set
+                    // kMaxLength. kMaxLength is set to a very high value ~ 1 Billion, so we
+                    // don't expect to hit this limit.
+                    let fn_name = name.into_js(&ctx).unwrap(); //  TODO: Check if this unwrap fails
```
Comment (tusharmath) [inline src/cli/javascript/runtime.rs:109]: no unwraps

File: src/cli/javascript/request_filter.rs (hunk 1 above)
Comment (tusharmath) [reply src/cli/javascript/request_filter.rs:24]: Worth asking on the repo. Wonder how the aws llrt team is doing this.

Comment (mayant15) [reply src/cli/javascript/request_filter.rs:24]: There's a macro for exposing classes that also adjusts prototype chains for you, but not for pure simple objects. llrt uses both, the [macro](https://github.com/awslabs/llrt/blob/63e705b023e00312c7b295fae1ed3625b0e0205d/src/events.rs#L382) and [manual trait implementations](https://github.com/awslabs/llrt/blob/63e705b023e00312c7b295fae1ed3625b0e0205d/src/buffer.rs#L23).

**Review by meskill — COMMENTED** [review]
for some reason these changes not compiling for me when using rust v1.75.0. And with v1.77 it's fine

File: src/cli/javascript/js_request.rs [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
+use rquickjs::{FromJs, IntoJs};
 use serde::{Deserialize, Serialize};
 
 use super::create_header_map;
 use crate::is_default;
+
 #[derive(Serialize, Deserialize, Debug)]
 #[serde(rename_all = "camelCase")]
 pub struct JsRequest {
```
Comment (tusharmath) [inline src/cli/javascript/js_request.rs:12]: I think a better structure for JsRequest would be:

```rust
pub struct JsRequest(reqwest::Request);
```

We can then implement into_js and from_js directly this way, which would improve performance quite considerably.

Comment (mayant15) [reply src/cli/javascript/js_request.rs:12]: I can (probably) do that, but to be clear I will *not* be able to use the macro for it. I will implement the `JsClass` myself. The macro assumes everything inside the class is a `JsClass` as well, and in particular that requires implementing Clone, which `reqwest::Request` doesn't.

https://github.com/DelSkayn/rquickjs/blob/c6e314e84587712af9808bb1bc45222f404caf8a/macro/src/class.rs#L368

Comment (tusharmath) [reply src/cli/javascript/js_request.rs:12]: Let's ignore the macros for now. It doesn't seem to be straight forward.

**Review by tusharmath — CHANGES_REQUESTED** [review]

**Review by tusharmath — APPROVED** [review]

**Pull request merged**